File: CCC-school-work/comporg/pnc2/kbeykirc/go/primeregba.go
// prime number calculator
// 
// to compile, type: go build primeregba.go
// to run (after compiling): ./primeregba QTY
//	where QTY is the number of primes you want to calculate

package main

import "fmt"
import "os"
import "strconv"
import "time"

func main() {

	var qty int // quantity of primes to calculate
	var err error // for checking atoi down below

	// check arguments and make sure quantity is ok
	if len(os.Args) == 2 {
		qty, err = strconv.Atoi(os.Args[1])

		if err != nil {
			fmt.Fprintf(os.Stderr, "Error in string conversion!\n")
			os.Exit(4)
		}

		if qty < 0 {
			fmt.Fprintf(os.Stderr, "Quantity must be positive\n")
			os.Exit(1)
		}
	} else {
		fmt.Fprintf(os.Stderr, "USAGE: ./primeregba QTY\n")
		os.Exit(2)
	}

	start := time.Now() // begin timing

	num := 2 // numbers to check for primality
	factor := 2 // start checking factors at 2
	num_primes := 0 // how many primes have we found?
	prime := 1 // is a number prime or not 

	for num_primes < qty {
		// check factors up to square root of number
		for factor*factor <= num {
			// if factor divides evenly, num is composite
			if num%factor == 0 {
				prime = 0
				break
			}

			factor += 1 // go to next factor
		}
		// if a number is prime, print it and increment num_primes
		if prime == 1 {
			num_primes += 1
			fmt.Fprintf(os.Stderr, "%d ", num)
		} else {
			prime = 1
		}

		factor = 2 // restore to starting value
		num += 1 // go to next number
	}

	fmt.Fprintf(os.Stderr, "\n")

	end := time.Now() // end time 

	diff := end.Sub(start) // get elapsed time

	fmt.Fprintf(os.Stdout, "%8.4f\n", diff.Seconds())
}

File: CCC-school-work/comporg/pnc2/kbeykirc/lua/primeregba.lua
--[[ prime number calculator 
--
--	to run, type: lua5.3 primeregba.lua QTY
--		where QTY is how many primes you want to calculate
--]]

-- check arguments and see if qty is positive
if #arg == 1 then
	qty = tonumber(arg[1])

	if qty < 0 then
		io.write("Must provide positive quantity!\n")
		os.exit()
	end
else
	io.write("USAGE: Must provide quantity as command line argument!\n")
	os.exit()
end

st_time = os.clock() -- start time

num_primes = 0 -- number of primes to calculate
num = 2 -- number to start checking for primality 
factor = 2 -- factor to test num
prime = 1 -- is a number prime or not

-- loop until we find all the primes we want
while num_primes < qty do
	-- check up to square root of num
	while factor*factor <= num do
		-- if factor divides evenly, num is composite
		if num%factor == 0 then
			prime = 0
			break
		end

		factor = factor + 1
	end
	-- if prime, increment num_primes and print num
	if prime == 1 then
		io.stderr:write(string.format("%d ", num))
		num_primes = num_primes + 1
	else
		prime = 1
	end

	num = num +  1 -- go to next number
	factor = 2 -- restore to default
end

io.stderr:write("\n")

end_time = os.clock() -- end time

io.write(string.format("%8.4f\n", end_time - st_time)) -- print elapsed time
